### PR TITLE
Fix #1115 -Dropbox working

### LINKED ISSUE #1115: App Crashes by Clicking on Accounts Button
**Actual Behaviour**

The App Crashes on Clicking Accounts Buttns

**Expected Behaviour**

The App should work fine

**Steps to reproduce it**

Open Phimpme and click on Accounts Button

**LogCat for the issue**

09-12 22:18:00.212 427-427/org.fossasia.phimpme D/AndroidRuntime: Shutting down VM
09-12 22:18:00.213 427-427/org.fossasia.phimpme E/AndroidRuntime: FATAL EXCEPTION: main
                                                                  Process: org.fossasia.phimpme, PID: 427
                                                                  java.lang.NoClassDefFoundError: Failed resolution of: Lcom/dropbox/client2/session/AppKeyPair;
                                                                      at org.fossasia.phimpme.accounts.AccountActivity.setupDropBox(AccountActivity.java:159)
                                                                      at org.fossasia.phimpme.accounts.AccountActivity.onCreate(AccountActivity.java:153)
                                                                      at android.app.Activity.performCreate(Activity.java:5953)
                                                                      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1128)
                                                                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2267)
                                                                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2388)
                                                                      at android.app.ActivityThread.access$800(ActivityThread.java:148)
                                                                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1292)
                                                                      at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                      at android.os.Looper.loop(Looper.java:135)
                                                                      at android.app.ActivityThread.main(ActivityThread.java:5312)
                                                                      at java.lang.reflect.Method.invoke(Native Method)
                                                                      at java.lang.reflect.Method.invoke(Method.java:372)
                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:901)
                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:696)
                                                                   Caused by: java.lang.ClassNotFoundException: Didn't find class "com.dropbox.client2.session.AppKeyPair" on path: DexPathList[[zip file "/data/app/org.fossasia.phimpme-2/base.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_dependencies_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_0_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_1_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_2_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_3_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_4_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_5_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_6_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_7_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_8_apk.apk", zip file "/data/app/org.fossasia.phimpme-2/split_lib_slice_9_apk.apk"],nativeLibraryDirectories=[/data/app/org.fossasia.phimpme-2/lib/arm, /vendor/lib, /system/lib]]
                                                                      at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                                      at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                                                                      at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                                                                      at org.fossasia.phimpme.accounts.AccountActivity.setupDropBox(AccountActivity.java:159) 
                                                                      at org.fossasia.phimpme.accounts.AccountActivity.onCreate(AccountActivity.java:153) 
                                                                      at android.app.Activity.performCreate(Activity.java:5953) 
                                                                      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1128) 
                                                                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2267) 
                                                                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2388) 
                                                                      at android.app.ActivityThread.access$800(ActivityThread.java:148) 
                                                                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1292) 
                                                                      at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                      at android.os.Looper.loop(Looper.java:135) 
                                                                      at android.app.ActivityThread.main(ActivityThread.java:5312) 
                                                                      at java.lang.reflect.Method.invoke(Native Method) 
                                                                      at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:901) 
                                                                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:696) 
                                                                  	Suppressed: java.lang.ClassNotFoundException: com.dropbox.client2.session.AppKeyPair
                                                                      at java.lang.Class.classForName(Native Method)
                                                                      at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                                                                      at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                                                                      at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                                                                      		... 16 more
                                                                   Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available

**Screenshots of the issue**

screenshot_2017-09-12-22-18-34

**Would you like to work on the issue?**

YES

### PR DESCRIPTION
**Pull request by codedsun**
Fix #1115 Dropbox sharing with other functionalities working.

Changes: Integrated CloudRail API Service.

Screenshots for the change:

### CODE REVIEW

**Comment by codedsun** [comment]
@mohitmanuja @anantprsd5 Kindly review.

**Comment by codedsun** [comment]
@anantprsd5 issues resolved,Review!

**Comment by codedsun** [comment]
@mohitmanuja chck this!

**Comment by anantprsd5** [comment]
@codedsun Please remove the API key

**Comment by codedsun** [comment]
@anantprsd5 will be updating the PR and adding the instructions till midnight..

**Comment by codedsun** [comment]
@anantprsd5 
Follow these
1.Signup at https://cloudrail.com/signup
2.Create an App on this cloudrail
3.https://www.dropbox.com/developers/apps go the phimpe old app and under the OAuth2 set the redirectURI to https://auth.cloudrail.com/org.fossasia.phimpme
4.Set the app key in Constants.CloudRailApp Key
Ping me if u stuck somewhere!

**Comment by codedsun** [comment]
@mohitmanuja Merge this!

File: app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java [hunk 1]
```diff
@@ -102,6 +101,7 @@
     private static final int OWNCLOUD_REQUEST_CODE = 9;
     private static final int RESULT_OK = 1;
     private static final int RC_SIGN_IN = 9001;
+    public final static String CLOUDRAIL_APP_KEY = "59c956f8c8b6755b444e4595";
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java]: @codedsun Please remove your API key

Comment (codedsun) [reply app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java]: Ok !Updating ! KIndly make the cloudrail account :)

**Comment by codedsun** [comment]
@mohitmanuja @anantprsd5 I have updated the PR, kindly chck it !

**Comment by codedsun** [comment]
@mohitmanuja Merge this too!

**Comment by anantprsd5** [comment]
@codedsun Working fine. Thanks!

File: app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java (hunk 1 above)
Comment (anantprsd5) [reply app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java]: Kindly remove your app key so that I can merge this.

**Comment by codedsun** [comment]
@anantprsd5 Thanks..will be updating.
My system is actually broken and is not working above 5-10 minutes.
I am not able to work currently until it's switched.

I will try to update the PR in 5-10 mins

**Pull request merged**